src/encode.rs

```diff
@@ -186,6 +187,41 @@ impl<'p> Serialize for SerializePyObject {
             let mut dt: SmallVec<[u8; 32]> = SmallVec::with_capacity(32);
             write_time(self.ptr, self.opts, &mut dt);
             serializer.serialize_str(str_from_slice!(dt.as_ptr(), dt.len()))
+        } else if is_type!(obj_ptr, UUID_PTR) && (self.opts & SERIALIZE_UUID == SERIALIZE_UUID) {
+            // In Python, `self.int` is the 128-bit integer value of the UUID;
+            // we can assume this will not fail, as tested in `test_uuid.py`
+            let py_int = ffi!(PyObject_GetAttr(self.ptr, INT_ATTR_STR));
+            // Copied in from https://github.com/PyO3/pyo3/blob/fb17d5e82f302f09b6611ac608edd1ce37504703/src/types/num.rs#L95
+            // because we don't have a `pyo3::Python` reference. However, because
+            // we haven't yet Py_DECREF'd the `py_int` attribute, the reference
+            // to `self.int` should be valid, and this should be safe to do.
+            // We know it's a `PyLongObject` because `self.int` is a 128-bit int,
+            // and know that _PyLong_AsByteArray won't error, as tested in test_uuid.py
+            let buffer: [c_uchar; 16] = [0; 16];
+            unsafe {
+                pyo3::ffi::_PyLong_AsByteArray(
+                    py_int as *mut pyo3::ffi::PyLongObject,
+                    buffer.as_ptr() as *const c_uchar,
+                    16,
+                    1, // Return a little-endian array
+                    0, // Unsigned - UUIDs can't be negative
+                )
+            };
+            ffi!(Py_DECREF(py_int));
+            let value = u128::from_le_bytes(buffer);
+            let hexadecimal = format!("{:032x}", value);
```

Comment:
With the `std` feature on SmallVec, this can avoid a heap allocation with:

```
    let mut hexadecimal: SmallVec<[u8; 32]> = SmallVec::with_capacity(32);
    write!(hexadecimal, "{:032x}", value).unwrap();
```